crates/extensions/c8y_mapper_ext/src/operations/firmware_update.rs

```diff
@@ -162,6 +116,56 @@ impl CumulocityConverter {
             Some(command.into_generic_command(&self.mqtt_schema)),
         ))
     }
+}
+
+pub fn firmware_update_topic_filter(mqtt_schema: &MqttSchema) -> TopicFilter {
```

Comment:
Minor: Same comment as in `config_snaphot.rs`. Better to restore this function and the next impl block to its former location to reduce the diff size, and to better highlight the changed bits.

Reply:
Indeed, the diff looks much easier to review. I'll remember not to shuffle the code next time for easier review.
fixed in b8bd269